catalogue/webapp/components/ImageEndpointSearchResults/ImageEndpointSearchResults.tsx

```diff
@@ -1,57 +1,120 @@
-import { FunctionComponent, useState } from 'react';
+import { FunctionComponent, useEffect, useState, useCallback } from 'react';
+import Gallery from 'react-photo-gallery';
+import styled from 'styled-components';
+
+import { Image, CatalogueResultsList } from '@weco/common/model/catalogue';
+
 import ExpandedImage from '../ExpandedImage/ExpandedImage';
 import ImageCard from '../ImageCard/ImageCard';
-import { Image, CatalogueResultsList } from '@weco/common/model/catalogue';
 import Modal from '@weco/common/views/components/Modal/Modal';
+import { convertImageUri } from '@weco/common/utils/convert-image-uri';
 
 type Props = {
   images: CatalogueResultsList<Image>;
 };
 
+const imageMargin = 16;
+
+const GalleryContainer = styled.div`
+  margin: 0 -${imageMargin / 2}px;
+
+  // This div is rendered by Gallery so we're targetting it on mobile only to ensure images are aligned on both sides
+  .react-photo-gallery--gallery > div {
+    justify-content: space-around;
+  }
+
+  @media (min-width: ${props => props.theme.sizes.medium}px) {
+    margin: 0 -${imageMargin}px;
+
+    .react-photo-gallery--gallery > div {
+      justify-content: flex-start;
+    }
+  }
+`;
+const ImageContainer = styled.li`
+  margin: 0 ${imageMargin ? imageMargin / 2 : 0}px
+    ${imageMargin ? imageMargin / 2 : 0}px;
+
+  @media (min-width: ${props => props.theme.sizes.medium}px) {
+    margin: 0 ${imageMargin}px ${imageMargin}px;
+  }
+`;
+
 const ImageEndpointSearchResults: FunctionComponent<Props> = ({
   images,
 }: Props) => {
   const [expandedImage, setExpandedImage] = useState<Image | undefined>();
+  const [data, setData] = useState([]) as any;
```

Comment:
Are you using an `Image[]` for `data` here, or am I misreading the code further down?

i.e.

```suggestion
  const [data, setData] = useState<Image[]>([]);
```

Reply:
Looked it up again, I have to extend Image to allow for `src/width/height` which are expected by `Gallery` (the library). I'll do that!

Reply:
They don't already have `width`/`height` available?! 😮

Reply:
@alexwlchan just a note here that we currently get the aspect ratio in an `inferrer` by looking at the actual image file, because this was the path of least resistance when the field was added (quite a long time ago).

What I think you were probably expecting, and what we should really aim to be doing (at least imo) soonish, is getting the `width`, `height` and `aspectRatio` from the IIIF `info.json` and sticking them all on the `Image`.